dealroom_firestore_connector/__init__.py

```diff
@@ -275,8 +294,11 @@ def set_history_doc_refs(
     """
 
     history_col = db.collection(HISTORY_COLLECTION_PATH)
+    
+    # Setting SERVER_TIMESTAMP for 'timestamp' key
+    parsed_payload = _parse_payload(payload, 'timestamp', firestore.SERVER_TIMESTAMP)
```

Comment:
use parsed_payload variable instead, which is the result of the parsing of the original payload

Reply:
All these versions of payload got confused me. I would start here by defining the `_payload` variable and would continue with this in the following transformations